src/datatypes/src/vectors/helper.rs

```diff
@@ -180,7 +181,7 @@ impl Helper {
                 ConstantVector::new(Arc::new(DateVector::from(vec![v])), length)
             }
             ScalarValue::Date64(v) => {
-                ConstantVector::new(Arc::new(DateTimeVector::from(vec![v])), length)
+                ConstantVector::new(Arc::new(TimestampMillisecondVector::from(vec![v])), length)
```

Comment:
In datafusion, this is in millisecond. Maybe we need to convert it into microsecond or return an error.
https://docs.rs/datafusion/latest/datafusion/common/enum.ScalarValue.html#variant.Date64